rsearch/config.go

```diff
@@ -0,0 +1,48 @@
+package rsearch
+
+import (
+	"gopkg.in/gcfg.v1"
+)
+
+// Done is an alias for empty struct, used for termination channels
```

Comment:
It's a pattern of sending broadcast message by means of closing a shared channel.

There are no process management and only 2 ways of communicating. Shared state variables or channels. Shared state variables are bad so channels if obvious way. The problem with channels is that you can't send message to all gorotines listening on a channel.

Termination channel or "done" channel is a pattern of creating a channel of **nil** types and have arbitrary number of goroutines sitting blocked on this channel forever. When you want to send message you close the channel and all goroutines receive broadcast message that channel was closed.